src/Sarif/Writers/SarifConsolidator.cs

```diff
@@ -0,0 +1,259 @@
+
+            // Truncate long Messages if configured
+            if (this.MessageLengthLimitBytes > 0)
+            {
+                if (result?.Message?.Text?.Length > this.MessageLengthLimitBytes)
+                {
+                    result.Message.Text = result.Message.Text.Substring(0, this.MessageLengthLimitBytes.Value);
+                }
+            }
+
+            // Trim Location and RelatedLocation properties
+            result.Locations = Trim(result.Locations);
+            result.RelatedLocations = Trim(result.RelatedLocations);
+
+            // Trim and Consolidate CodeFlows and ThreadFlows
+            ConsolidateCodeFlows(result);
+        }
+
+        public void ConsolidateCodeFlows(Result result)
+        {
+            if (result.CodeFlows != null)
+            {
+                foreach (CodeFlow cFlow in result.CodeFlows)
+                {
+                    foreach (ThreadFlow tFlow in cFlow.ThreadFlows)
+                    {
+                        Consolidate(tFlow);
+                    }
+                }
+            }
+        }
+
+        public void Consolidate(ThreadFlow tFlow)
+        {
+            if (tFlow.Locations != null)
+            {
+                TotalThreadFlowLocations += tFlow.Locations.Count;
+
+                for (int i = 0; i < tFlow.Locations.Count; ++i)
+                {
+                    ThreadFlowLocation tfl = tFlow.Locations[i];
+                    Trim(tfl.Location);
+
+                    if (tfl.Index == -1)
+                    {
+                        if (!_uniqueThreadFlows.TryGetValue(tfl, out int tflIndex))
+                        {
+                            tflIndex = _run.ThreadFlowLocations.Count;
+                            _run.ThreadFlowLocations.Add(tfl);
+                            _uniqueThreadFlows[tfl] = tflIndex;
+                        }
+
+                        tFlow.Locations[i] = new ThreadFlowLocation() { Index = tflIndex };
+                    }
+                }
+            }
+        }
+
+        public IList<Location> Trim(IList<Location> locations)
+        {
+            if (locations == null || locations.Count == 0) { return locations; }
+
+            HashSet<Location> uniqueLocations = new HashSet<Location>(Location.ValueComparer);
+            List<Location> newLocations = new List<Location>();
+
+            this.TotalLocations += locations.Count;
+            for (int i = 0; i < locations.Count; ++i)
+            {
+                Location loc = locations[i];
+
+                // Trim Locations themselves
+                Trim(loc);
+
+                // Keep only one copy of each unique location
+                if (uniqueLocations.Add(loc))
+                {
+                    this.UniqueLocations++;
+                    newLocations.Add(loc);
+                }
+            }
+
+            return newLocations;
+        }
+
+        public void Trim(Location loc)
```

Comment:
>loc [](start = 34, length = 3)

certainly a shortened name shouldn't be used for a public method, let's use 'location'. i suspect VS method extraction created this parameter name for you (based on your local from the originating caller.

Reply:
I don't like how wordy and long very long argument names make the method body. I want the reader to focus on what I'm doing with the instance.

I see how since you like named parameters you want the name to be more descriptive (because otherwise the calling code will never show it).

I've changed it. This is a style issue where we disagree and should talk further at some point.